src/config.rs

```diff
@@ -39,33 +51,194 @@ pub struct Config {
     #[builder(default)]
     pub rule: Option<String>,
 
-    rules: Vec<Arc<dyn Rule>>,
+    all_rules: Vec<Arc<dyn Rule>>,
 
     #[builder(default)]
     pub fix: bool,
 
     #[builder(default)]
     pub report_fixed_violations: bool,
+
+    #[builder(default)]
+    pub config_file_path: Option<PathBuf>,
+
+    pub rule_configurations: Vec<RuleConfiguration>,
 }
 
 impl Config {
-    fn rules(&self) -> &[Arc<dyn Rule>] {
-        &self.rules
+    fn get_active_rules(&self) -> Vec<Arc<dyn Rule>> {
+        self.rule_configurations
+            .iter()
+            .filter(|rule_config| rule_config.level != ErrorLevel::Off)
+            .map(|rule_config| {
+                self.all_rules
+                    .iter()
+                    .find(|rule| rule.meta().name == rule_config.name)
+                    .unwrap_or_else(|| panic!("Unknown rule: '{}'", rule_config.name))
+                    .clone()
+            })
+            .collect()
+    }
+
+    fn filter_based_on_rule_argument(
+        &self,
+        active_rules: Vec<Arc<dyn Rule>>,
+    ) -> Vec<Arc<dyn Rule>> {
+        match self.rule.as_ref() {
+            Some(rule_arg) => {
+                let filtered = active_rules
+                    .into_iter()
+                    .filter(|rule| &rule.meta().name == rule_arg)
+                    .collect::<Vec<_>>();
+                if !filtered.is_empty() {
+                    return filtered;
+                }
+                self.rule_argument_error();
+            }
+            None => active_rules,
+        }
+    }
+
+    fn rule_argument_error(&self) -> ! {
+        let rule_arg = self.rule.as_ref().unwrap();
+        if self
+            .all_rules
+            .iter()
+            .any(|rule| &rule.meta().name == rule_arg)
+        {
+            panic!("The '{rule_arg}' rule is configured as inactive");
+        } else {
+            panic!("Unknown rule: '{rule_arg}'");
+        }
     }
 
     pub fn get_instantiated_rules(&self) -> Vec<InstantiatedRule> {
-        let instantiated_rules = self
-            .rules()
+        let active_rules = self.get_active_rules();
+        if active_rules.is_empty() {
+            panic!("No configured active rules");
+        }
+        let active_rules = self.filter_based_on_rule_argument(active_rules);
+        let instantiated_rules = active_rules
             .into_iter()
             .map(|rule| InstantiatedRule::new(rule.clone(), self))
-            .filter(|rule| match self.rule.as_ref() {
-                Some(rule_arg) => &rule.meta.name == rule_arg,
-                None => true,
-            })
             .collect::<Vec<_>>();
         if instantiated_rules.is_empty() {
             panic!("Invalid rule name: {:?}", self.rule.as_ref().unwrap());
         }
         instantiated_rules
     }
 }
+
+impl ConfigBuilder {
+    pub fn default_rule_configurations(&mut self) -> &mut Self {
```

Comment:
This was my take on making it ergonomic for eg the tests to build a `Config` - they can say:
```
.all_rules([some_rule_under_test()])
.default_rule_configurations()
...
```
to say "make all of these rules configured as `error`"